### PR TITLE
Document potential in-place upgrade process for managed instances

### PR DESCRIPTION
**Pull request by caugustus-sourcegraph**
Steps to run through an in-place upgrade. I've been testing this on our low-risk managed instances (trial instances without data or active users) and the process is significantly faster and less impactful to any customers who might actually be on the site. 

However, there is the (small) risk of **data loss or downtime** if something goes wrong with the update, so I'm not sure how comfortable we would be applying this to our larger customer instances.

### CODE REVIEW

**Review by kevinwojo — APPROVED** [review]
Groovy

**Comment by michaellzc** [comment]
I haven't got the chance to go through the update process yet, but the instruction makes sense to me.

File: content/departments/product-engineering/engineering/cloud/delivery/managed/upgrade_process.md [hunk 1]
```diff
@@ -445,3 +452,85 @@ For reference, the above steps were adapted from [Working with persistent disks]
...
+Validate all variables are set:
+
+```sh
+./util/validate-env.ts
+```
+
+Then set up a branch for your changes:
+
+```sh
+git checkout -b $CUSTOMER/upgrade to $NEW_VERSION in-place
```
Comment (bobheadxi) [inline content/departments/product-engineering/engineering/cloud/delivery/managed/upgrade_process.md]:
```suggestion
git checkout -b $CUSTOMER/upgrade-$NEW_VERSION-in-place
```

(invalid branch name 😛 )

File: content/departments/product-engineering/engineering/cloud/delivery/managed/upgrade_process.md [hunk 2]
```diff
@@ -445,3 +452,85 @@ For reference, the above steps were adapted from [Working with persistent disks]
...
+# all the below steps are documented assuming you are in the customer deployment directory
+cd $CUSTOMER
+```
+
+### 1) Create a snapshot
+
+Create a snapshot:
+
+```sh
+../util/create-snapshot.ts $OLD_DEPLOYMENT upgrade-machine-$DATE
```
Comment (bobheadxi) [inline content/departments/product-engineering/engineering/cloud/delivery/managed/upgrade_process.md]:
```suggestion
../util/create-snapshot.ts $OLD_DEPLOYMENT
```

the default generated name for snapshots will work here, same as https://github.com/sourcegraph/handbook/blob/document-inplace-upgrades/content/departments/product-engineering/engineering/cloud/delivery/managed/upgrade_process.md#3-create-a-snapshot-of-the-current-deployment

**Review by bobheadxi — COMMENTED** [review]
noticed a few things, but otherwise looks slick 😁

File: content/departments/product-engineering/engineering/cloud/delivery/managed/upgrade_process.md (hunk 2 above)
Comment (caugustus-sourcegraph) [reply content/departments/product-engineering/engineering/cloud/delivery/managed/upgrade_process.md]: TY! Bad copy-paste + editing on my part.

**Pull request merged**